graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/config/DXGraphQLConfig.java

```diff
@@ -71,6 +75,17 @@ public void updated(String pid, Dictionary<String, ?> properties) throws Configu
 
         // parse properties
         Enumeration<String> keys = properties.keys();
+        boolean isDefaultConfig = properties.get("felix.fileinstall.filename") != null &&
+                properties.get("felix.fileinstall.filename").toString().endsWith("org.jahia.modules.graphql.provider-default.cfg");
+
+        /*
+         * Keep track of ENABLE_INTROSPECTION_MODE property if it's defined or not (using null) in the default configuration
+         * and only set it to a true/false value if it's defined in the configuration file
+         */
+        if (isDefaultConfig) {
+            introspectionEnabled = null;
```

Comment:
The graphql-dxm-provider configuration contains more than just introspection couldn't this logic be problematic, for example if the user modified "graphql.fields.node.limit" ?